pkg/authorization/api/types.go

```diff
@@ -131,6 +131,76 @@ type PolicyBinding struct {
 	RoleBindings map[string]RoleBinding `json:"roleBindings"`
 }
 
+type ResourceAccessReviewSpec struct {
+	// Verb is one of: get, list, watch, create, update, delete
+	Verb string `json:"verb"`
+	// Resource is one of the existing resource types
+	Resource string `json:"resource"`
+	// Content is the actual content of the request for create and update
+	Content kruntime.EmbeddedObject `json:"content,omitempty"`
+	// ResourceName is the name of the resource being requested for a "get" or deleted for a "delete"
+	ResourceName string `json:"resourceName,omitempty"`
+}
+
+// ResourceAccessReviewStatus describes who can perform the action
+type ResourceAccessReviewStatus struct {
+	// Users is the list of users who can perform the action
+	Users []string `json:"users"`
+	// Groups is the list of groups who can perform the action
+	Groups []string `json:"groups"`
+	// EvaluationError is optional.  It indicates why a ResourceAccessReview failed during evaluation
+	EvaluationError string `json:"evaluationError,omitempty"`
```

Comment:
api.Status is any failure.  If you don't need to return a structured error, you should be returning api.Status.

----- Original Message -----
> > +	// Resource is one of the existing resource types
> > +	Resource string `json:"resource"`
> > +	// Content is the actual content of the request for create and update
> > +	Content kruntime.EmbeddedObject `json:"content,omitempty"`
> > +	// ResourceName is the name of the resource being requested for a "get"
> > or deleted for a "delete"
> > +	ResourceName string `json:"resourceName,omitempty"`
> > +}
> > +
> > +// ResourceAccessReviewStatus describes who can perform the action
> > +type ResourceAccessReviewStatus struct {
> > +	// Users is the list of users who can perform the action
> > +	Users []string `json:"users"`
> > +	// Groups is the list of groups who can perform the action
> > +	Groups []string `json:"groups"`
> > +	// EvaluationError is optional.  It indicates why a ResourceAccessReview
> > failed during evaluation
> > +	EvaluationError string `json:"evaluationError,omitempty"`
> 
> > A failure to evaluate should be returned as an api.Status, unless I'm
> > missing a reason for an optional error message.
> 
> api.Status felt like it would indicate that a failure happened while
> marshalling the values to make the authorization determination as opposed to
> an error while actually evaluating against policy.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/800/files#r24452496